### PR TITLE
Tests for the TutorMatching screen

### PR DESCRIPTION
**Pull request by LeaBlancher**
# Testing of the TutorMatching screen

### PR Description
This PR add some tests for the TutorMatching screen. It covers both the use of this screen when a student choose a tutor after the creation of a lesson and then he chooses a tutor after some tutors propose themselves.

Note that this PR is made by a common work of @alexs-rufer and @LeaBlancher. It is made on the base of the code of PR #107 that adds the description field to the tutor profiles.

### Tests Description
- **testNoTutorsMessageDisplayed_whenNoMatchingTutors()** => It tests that in case no tutors are available for the lesson (due to the matching algorithm), the correct message is displayed.
- **everythingIsDisplayed()** => This test if every components (top app bar, tutors informations, confirmation button) are displayed when a tutor is available for the lesson.
- **testConfirmButton_whenNoTutorSelected()** => It tests if the lesson is correctly saved and that we navigate back to the home screen when we confirm the lesson without selecting any tutor.
- **selectAndConfirmLessonWithTutor()** => It tests if the dialog is well displayed when we select a tutor and that we can correctly confirm our choice with the dialog confirmation button,
- **selectThenDismissLessonWithTutor()** => This test select a tutor, check if the dialog box is displayed and then cancel his action to test if this cancellation works properly.
- **goBackButton_navigateBack()** => This test click on the goBack button (in top app bar) to see if it acts as intented.
- **confirmButtonNotDisplayed_whenStatusIsNotMatching()** => It checks that when the screen is used by a student to select a tutor that has proposed himself for the lesson, the confirmation button is not displayed (because the lesson is already in the database)

### Technical details
In order to test the screen, we used a mocking of the lesson and profile repository and view model with correct handling of the functions that were supposed to be used for the screen to work as intented.
We also add some testTag in the TutorMatching screen to simplify testing of the different composables.

### Note for PR review
Since this PR is made on base of PR #107, it has some updates concerning the description field that will be reviewed in this other PR, but since it is not merge yet, it appears as "new" code. The only files that contain really new code in this PR are TutorMatchingTest.kt (for the tests) and TutorMatching.kt (for the testTag).

### CODE REVIEW

File: app/src/androidTest/java/com/github/se/project/ui/lesson/TutorMatchingTest.kt [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+
+    whenever(profilesRepository.getProfiles(any(), any())).thenAnswer { invocation ->
+      val onSuccess = invocation.arguments[0] as (List<Profile>) -> Unit
+      onSuccess(tutorsFlow.value) // Simulate returning an empty list of profiles
+    }
+
+    whenever(lessonRepository.addLesson(any(), any(), any())).thenAnswer { invocation ->
+      val onSuccess = invocation.arguments[1] as () -> Unit
+      onSuccess() // Simulate a successful update
+    }
```
Comment (MPepito) [inline app/src/androidTest/java/com/github/se/project/ui/lesson/TutorMatchingTest.kt:133]: Good job adding the correct mocking and response of them

File: app/src/androidTest/java/com/github/se/project/ui/lesson/TutorMatchingTest.kt [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+
+  @Before
+  fun setup() {
+    // Mock dependencies
+    profilesRepository = mock(ProfilesRepository::class.java)
+    lessonRepository = mock(LessonRepository::class.java)
+    navController = mock()
+
+    listProfilesViewModel = ListProfilesViewModel(profilesRepository)
+    listProfilesViewModel = spy(listProfilesViewModel)
```
Comment (MPepito) [inline app/src/androidTest/java/com/github/se/project/ui/lesson/TutorMatchingTest.kt:102]: Well done using spy on the viewmodel

File: app/src/androidTest/java/com/github/se/project/ui/lesson/TutorMatchingTest.kt [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+
+    lessonViewModel = LessonViewModel(lessonRepository)
+    lessonViewModel = spy(lessonViewModel)
+
+    navigationActions = NavigationActions(navController)
+    navigationActions = spy(navigationActions)
+
+    // Stub navigation actions
+    doNothing().`when`(navigationActions).navigateTo(anyString())
```
Comment (MPepito) [inline app/src/androidTest/java/com/github/se/project/ui/lesson/TutorMatchingTest.kt:111]: we could test difference response from the mocked Navhost but for the moment it is enough to test our limited feature that we implemented

**Review by MPepito — APPROVED** [review]
Clean implementation and construction of the test making it easy to understand what is tested. Any of us will be able to quickly write new test based on the current test, once we implement new feature in MS3.

**Review by A-M-00 — APPROVED** [review]
Looks good to me

**Review by aferrante1 — APPROVED** [review]

**Pull request merged**